app/script/entity/message/MemberMessage.js

```diff
@@ -0,0 +1,156 @@
+/*
+ * Wire
+ * Copyright (C) 2017 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+'use strict';
+
+window.z = window.z || {};
+window.z.entity = z.entity || {};
+
+z.entity.MemberMessage = class MemberMessage extends z.entity.SystemMessage {
+  constructor() {
+    super();
+
+    this.super_type = z.message.SuperType.MEMBER;
+    this.member_message_type = z.message.SystemMessageType.NORMAL;
+
+    this.user_ets = ko.observableArray();
+    this.user_ids = ko.observableArray();
+
+    // Users joined the conversation without sender
+    this.joined_user_ets = ko.pureComputed(() => {
+      return (Array.from(this.user_ets())
+        .filter((user_et) => user_et.id !== this.user().id)
+        .map((user_et) => user_et)
+      );
+    });
+
+    // Users joined the conversation without self
+    this.remote_user_ets = ko.pureComputed(() => {
+      return (Array.from(this.user_ets())
+        .filter((user_et) => !user_et.is_me)
+        .map((user_et) => user_et)
+      );
+    });
+
+    this._generate_name_string = (declension) => {
+      if (declension == null) {
+        declension = z.string.Declension.ACCUSATIVE;
+      }
+      return z.util.LocalizerUtil.join_names(this.joined_user_ets(), declension);
+    };
+
+    this._get_caption_connection = function(connection_status) {
+      let identifier;
+      switch (connection_status) {
+        case z.user.ConnectionStatus.BLOCKED:
+          identifier = z.string.conversation_connection_blocked;
+          break;
+        case z.user.ConnectionStatus.SENT:
+          return '';
+        default:
+          identifier = z.string.conversation_connection_accepted;
+      }
+      return z.localization.Localizer.get_text(identifier);
+    };
+
+    this._get_caption_with_names = (key, declension) => {
+      return z.localization.Localizer.get_text({
+        id: key,
+        replace: {
+          placeholder: '%@names',
+          content: this._generate_name_string(declension),
+        },
+      });
+    };
+
+    this.show_large_avatar = () => {
+      const large_avatar_types = [
+        z.message.SystemMessageType.CONNECTION_ACCEPTED,
+        z.message.SystemMessageType.CONNECTION_REQUEST,
+      ];
+      return Array.from(large_avatar_types).includes(this.member_message_type);
```

Comment:
No need for array.from on array

`return large_avatar_types.includes(this.member_message_type);`